common/git-hash/src/lib.rs

```diff
@@ -53,3 +93,93 @@ fn opt_tokenize(s: Option<String>) -> Result<TokenStream, LexError> {
         .unwrap_or_else(|| "None".to_owned())
         .parse()
 }
+
+struct InfoMoveStdlib {
+    version: Option<String>,
+    branch: Option<String>,
+    hash: Option<String>,
+}
+impl InfoMoveStdlib {
+    /// Get move-stdlib information from Cargo.lock
+    /// return (Version, Branch, Hash)
+    pub fn from_cargo_lock() -> InfoMoveStdlib {
+        fn parse_version(version: &str) -> String {
+            version[version.find('"').unwrap_or(10)..]
+                .trim_matches('"')
+                .to_string()
+        }
+        /// return (Branch, Hash)
+        fn parse_source(source: &str) -> (Option<String>, Option<String>) {
+            let source = source[source.find('"').unwrap_or(9)..].trim_matches('"');
+            let branch = source.find("diem.git?branch=").map(|position| {
+                let mut branch = &source[position + 16..];
+                if let Some(position) = branch.find('&') {
+                    branch = &branch[..position]
+                } else if let Some(position) = branch.find('#') {
+                    branch = &branch[..position]
+                }
+                branch.to_string()
+            });
+            let hash = source.find('#').map(|position| {
+                let hash = &source[position + 1..];
+                { &hash[..7] }.to_string()
+            });
+            (branch, hash)
+        }
+
+        let toml_string = std::fs::read_to_string("Cargo.lock").unwrap_or_default();
+        toml_string
+            .split("[[package]]")
+            .find(|package| package.contains(r#"name = "move-stdlib""#))
```

Comment:
We need a version of our standard library, not diem.
See https://github.com/pontem-network/move-tools/blob/master/dove/src/cmd/init.rs#L20

Reply:
@RIg410 Thanks. Fixed it